### ISSUE TITLE
Feature Request: Allow permanently hiding Outline Pane, Status Bar, File Path bar and Minimap scrollbar in Editor while "Maximize Current Pane" mode is active

### ISSUE THREAD

**Issue by manojbaishya** [spyder-ide/spyder#14530]
## Problem Description

When working in a RStudio-style layout in a small screen, it is desirable to put the editor in maximized mode and work. Upon switching off the outline pane from main menu while on the maximized mode, it disappears. But after restore and maximizing a second time, the outline pane reappears. In a laptop screen it looks very cluttered.

1. How can I permanently disable Outline Pane in maximized mode? The Symbol Finder "Ctrl + Alt + P" does the exact same thing and it is sufficient for me.
2. How can I disable the Status bar only in maximized mode?
3. Is it possible to hide the file path section also over the editor?

![Four-Pane](https://user-images.githubusercontent.com/28330014/103487403-25393b00-4e2b-11eb-8ace-0a42968c5154.png)

![Editor-Maximized-Cluttered](https://user-images.githubusercontent.com/28330014/103487271-2b7ae780-4e2a-11eb-96af-3947ab9ee7c4.png)

### What steps reproduce the problem?

1. Open Spyder and switch to a four-pane layout.
2. Focus on the Editor.
3. Press "Ctrl + Alt + Shift + M" or select View > Maximize Current Pane

### What is the expected output? What do you see instead?

I have been able to achieve a "Zen Mode" by manually disabling the Outline Pane (which is temporary) and Status Bar (global setting). Was not able to hide the file path section at the top.

![Pseudo-Zen-Mode](https://user-images.githubusercontent.com/28330014/103487327-80b6f900-4e2a-11eb-9a11-948ef6f52625.png)

For reference, this is the Zen Mode of Sublime Text.

![Screenshot_2021-01-04_01-02-11](https://user-images.githubusercontent.com/28330014/103487450-9ed12900-4e2b-11eb-8a39-ed258cf544ea.png)

Currently, I see the layout as depicted above in description.

## Versions

* Spyder version: 4.2.1 None
* Python version: 3.8.6 64-bit
* Qt version: 5.12.5
* PyQt5 version: 5.12.3
* Operating System: Linux 5.4.0-45-generic
* conda: 4.9.2

### Dependencies

```

# Mandatory:
atomicwrites >=1.2.0            :  1.4.0 (OK)
chardet >=2.0.0                 :  4.0.0 (OK)
cloudpickle >=0.5.0             :  1.6.0 (OK)
diff_match_patch >=20181111     :  20200713 (OK)
intervaltree >=3.0.2            :  3.1.0 (OK)
IPython >=7.6.0                 :  7.19.0 (OK)
jedi =0.17.2                    :  0.17.2 (OK)
jsonschema >=3.2.0              :  3.2.0 (OK)
keyring >=17.0.0                :  21.2.1 (OK)
nbconvert >=4.0                 :  6.0.7 (OK)
numpydoc >=0.6.0                :  1.1.0 (OK)
parso =0.7.0                    :  0.7.0 (OK)
pexpect >=4.4.0                 :  4.8.0 (OK)
pickleshare >=0.4               :  0.7.5 (OK)
psutil >=5.3                    :  5.7.2 (OK)
pygments >=2.0                  :  2.7.3 (OK)
pylint >=1.0                    :  2.6.0 (OK)
pyls >=0.36.2;<1.0.0            :  0.36.2 (OK)
pyls_black >=0.4.6              :  0.4.6 (OK)
pyls_spyder >=0.3.0             :  0.3.0 (OK)
qdarkstyle >=2.8                :  2.8.1 (OK)
qtawesome >=0.5.7               :  1.0.1 (OK)
qtconsole >=5.0.1               :  5.0.1 (OK)
qtpy >=1.5.0                    :  1.9.0 (OK)
rtree >=0.8.3                   :  0.9.4 (OK)
setuptools >=39.0.0             :  51.0.0.post20201207 (OK)
sphinx >=0.6.6                  :  3.4.0 (OK)
spyder_kernels >=1.10.1;<1.11.0 :  1.10.1 (OK)
textdistance >=4.2.0            :  4.2.0 (OK)
three_merge >=0.1.1             :  0.1.1 (OK)
watchdog >=0.10.3               :  0.10.4 (OK)
xdg >=0.26                      :  0.27 (OK)
zmq >=17                        :  19.0.2 (OK)

# Optional:
cython >=0.21                   :  0.29.21 (OK)
matplotlib >=2.0.0              :  3.3.2 (OK)
numpy >=1.7                     :  1.19.2 (OK)
pandas >=1.1.1                  :  1.1.5 (OK)
scipy >=0.17.0                  :  1.5.2 (OK)
sympy >=0.7.3                   :  1.7.1 (OK)

# Spyder plugins:
spyder_terminal                 :  0.5.0 (OK)
spyder_unittest                 :  0.4.1 (OK)
```

**Comment by steff456** [comment]
Hi @manojbaishya,

We are currently working on making changes to the interface, so this is really valuable input. I'm going to answer your questions and if you have any further question please don't hesitate to ask.

> 1. How can I permanently disable Outline Pane in maximized mode? The Symbol Finder "Ctrl + Alt + P" does the exact same thing and it is sufficient for me.

I'm not sure if you already tried closing the outline pane when you are in maximized mode. If you close the pane, it will not reappear unless you show it again in `View>Panes>Outline`.

> 2. How can I disable the Status bar only in maximized mode?

This is not possible right now for Spyder, why would you prefer to disable the status bar in maximized mode?

> 3. Is it possible to hide the file path section also over the editor?

This is not possible right now for Spyder, but if we could, would you hide it only in the maximized/zen view?

For now, I could recommend to close the panes that you are not working and using while using the editor.

**Comment by manojbaishya** [comment]
Hello,

Great news on the interface! Regarding the points I raised:

> 1. I'm not sure if you already tried closing the outline pane when you are in maximized mode. If you close the pane, it will not reappear unless you show it again in View>Panes>Outline.

Yes, I have tried closing it and true that it won't appear unless I go back to the menu to turn it on again.. But this is all while the editor is still maximized. Suppose, I am done editing code and need to *restore* the window to the four-pane layout to run the code, inspect variables and see plots, etc., and then I discover some changes are required; once I go back to maximized mode, the outline pane reappears! Below is a small video demonstration of this flow... (I am wondering if this behaviour is just on my system or for others also.) Also, since I don't have the outline pane displayed in the four-pane layout, the tickbox `View>Panes>Outline` is unchecked. However, in max mode since the pane is displayed by default, that checkbox has to be clicked twice to get it to disappear, this is visible in the video..

https://user-images.githubusercontent.com/28330014/103621850-d9d27a00-4f5b-11eb-95e5-6bc6005886c4.mp4

> 2. This is not possible right now for Spyder, why would you prefer to disable the status bar in maximized mode?

The information on the Status Bar can be obtained in the four-panel mode and hence if we need to see Line #, Col #, etc., `Restore Current Pane` from max mode is just a keyboard shortcut away.. My thoughts are that when someone maximizes the editor, they want to think about a problem and focus on the code and nothing else.. Sublime Text's Zen Mode captures this best! (but I can't use it since it is neither free nor suits data-oriented workflow). I could achieve it in Spyder like below but only after hiding the status bar totally (which disables it in four pane view too..)

![temporary-zen-mode](https://user-images.githubusercontent.com/28330014/103487327-80b6f900-4e2a-11eb-9a11-948ef6f52625.png)

> 3. This is not possible right now for Spyder, but if we could, would you hide it only in the maximized/zen view?

True. To me, hiding the file path only makes sense in the maximized mode. In the normal view, path information is valuable to have at a glance.

----------------------------------------------------------------------------------------------------------------------------------------------------------

> For now, I could recommend to close the panes that you are not working and using while using the editor.

I have attempted to close all panes that I don't need. This is the bare minimum layout I could make. Are there quick shortcuts for toggling Console, Variable Explorer, and Plots? Then I could close all and just have the editor in view.. and summon those panes when needed!

![Screenshot_2021-01-05_10-45-05](https://user-images.githubusercontent.com/28330014/103622906-5ade4100-4f5d-11eb-96d9-9653f37c9d1e.png)

----------------------------------------------------------------------------------------------------------------------------------------------------------

I am trying to take cues from VS Code's design (which I use for normal scripting) and feed that back here.. I understand that aesthetics are naturally personal but if many users give feedback, I believe the interface and UX will evolve holistically.

Also, I encountered another problem yesterday. When I changed the keyboard shortcuts of "Toggle Comment", "Block Comment" and "Switch to IPython Console" to match that of VS Code, the entries in the Search menu greyed out and those shortcuts were lost. There are two shortcut entries per item (go to `Tools > Preferences > Keyboard Shortcuts` and search for `find_replace` context). Only if I go back and set the 2nd one for each of those, were those functions restored. The menu is still greyed out but available via my shortcuts.

![image](https://user-images.githubusercontent.com/28330014/103624191-31beb000-4f5f-11eb-872d-ebfe056ace9f.png)
![image](https://user-images.githubusercontent.com/28330014/103624039-fa500380-4f5e-11eb-80fa-b5794b3b4daa.png)

----------------------------------------------------------------------------------------------------------------------------------------------------------
PS: I don't know in which version you added SVG support in Plots pane (VSCode and Pycharm do not have a zoomable svg plot collector pane), but that and the Variable Explorer got me to fully switch from a university-issued MATLAB to Spyder..  Last I used it in 2016 v3, it didn't have much going but in 2021 Spyder (and Jupyter) is the de facto scientific environment for Python... So amazing work indeed!

**Comment by steff456** [comment]
Thanks for all this information @manojbaishya! This is going to be really helpful for a lot of the decisions we are creating that will be available to future releases of Spyder. Right now we are working really hard to make possible for users to make Spyder more extensible and make some of the things you'll like to do possible.

**Comment by ccordoba12** [comment]
> How can I permanently disable Outline Pane in maximized mode? The Symbol Finder "Ctrl + Alt + P" does the exact same thing and it is sufficient for me.

Duplicate of issue #16265.

> How can I disable the Status bar only in maximized mode?

It's not possible and we're not planning to implement that feature because we have more pressing things to do, sorry,

> Is it possible to hide the file path section also over the editor?

No, it's not. and our reason for not planning to do it is the same as the above one.

**Comment by JohnRobson** [comment]
Removing the "file path" above the tabs would be good to save screen space. The full path is already shown when the mouse is over the tab, so, this static "file path" info is unnecessary.
